src/main/java/com/box/sdk/BoxWebHook.java

```diff
@@ -0,0 +1,685 @@
+import com.box.sdk.internal.utils.CollectionUtils;
+import com.box.sdk.internal.utils.CollectionUtils.Mapper;
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+
+/**
+ * Box WebHook resource.
+ *
+ * @author Stanislav Dvorscak
+ *
+ * @since 2.2.1
+ *
+ */
+public class BoxWebHook extends BoxResource {
+
+    /**
+     * JSON Key for {@link BoxWebHook} {@link #getID()}.
+     */
+    private static final String JSON_KEY_ID = "id";
+
+    /**
+     * JSON Key for {@link BoxWebHook.Info#getTarget()}.
+     */
+    private static final String JSON_KEY_TARGET = "target";
+
+    /**
+     * JSON Key for {@link BoxWebHook.Target#getType()}.
+     */
+    private static final String JSON_KEY_TARGET_TYPE = "type";
+
+    /**
+     * JSON Key for {@link BoxWebHook.Target#getId()}.
+     */
+    private static final String JSON_KEY_TARGET_ID = "id";
+
+    /**
+     * JSON Key for {@link BoxWebHook.Info#getAddress()}.
+     */
+    private static final String JSON_KEY_ADDRESS = "address";
+
+    /**
+     * JSON Key for {@link BoxWebHook.Info#getTriggers()}.
+     */
+    private static final String JSON_KEY_TRIGGERS = "triggers";
+
+    /**
+     * {@link URLTemplate} for {@link BoxWebHook}s resource.
+     */
+    private static final URLTemplate WEBHOOKS_URL_TEMPLATE = new URLTemplate("webhooks");
+    /**
+     * {@link URLTemplate} for single {@link BoxWebHook} resource.
+     */
+    private static final URLTemplate WEBHOOK_URL_TEMPLATE = new URLTemplate("webhooks/%s");
+
+    /**
+     * Maps a {@link Trigger} to its {@link Trigger#getValue()}.
+     */
+    private static final Mapper<String, BoxWebHook.Trigger> TRIGGER_TO_VALUE = new Mapper<String, BoxWebHook.Trigger>() {
+
+        @Override
+        public String map(Trigger input) {
+            return input.getValue();
+        }
+
+    };
+
+    private static final Mapper<BoxWebHook.Trigger, JsonValue> JSON_VALUE_TO_TRIGGER = new Mapper<Trigger, JsonValue>() {
+        @Override
+        public Trigger map(JsonValue value) {
+            return Trigger.fromValue(value.asString());
+        }
+    };
+
+    /**
+     * Constructor.
+     *
+     * @param api
+     *            {@link #getAPI()}
+     * @param id
+     *            {@link #getID()}
+     */
+    public BoxWebHook(BoxAPIConnection api, String id) {
+        super(api, id);
+    }
+
+    /**
+     * Adds a {@link BoxWebHook} to a provided {@link BoxItem}.
+     *
+     * @param target
+     *            {@link BoxItem} web resource
+     * @param address
+     *            {@link URL} where the notification should send to
+     * @param triggers
+     *            events this {@link BoxWebHook} is interested in
+     * @return created {@link BoxWebHook}
+     *
+     * @see #create(BoxItem, URL, Set)
+     */
+    public static BoxWebHook.Info create(BoxItem target, URL address, BoxWebHook.Trigger... triggers) {
```

Comment:
In future there might be support for other resources as target. E.g. User/Enterprise.
We should make it generic, that makes adding other resource types as target easier.